sdk/core/Azure.Core/src/Serialization/ModelJsonConverter.cs

```diff
@@ -57,7 +57,7 @@ public override IModelJsonSerializable<object> Read(ref Utf8JsonReader reader, T
 #pragma warning restore AZC0014 // Avoid using banned types in public API
         {
             using JsonDocument document = JsonDocument.ParseValue(ref reader);
-            return (IModelJsonSerializable<object>)ModelSerializer.Deserialize(BinaryData.FromString(document.RootElement.GetRawText()), typeToConvert, ModelSerializerOptions);
+            return (IModelJsonSerializable<object>)ModelSerializer.Deserialize(BinaryData.FromString(document.RootElement.GetRawText()), typeToConvert, ModelSerializerOptions)!;
```

Comment:
QQ: Do we actually know that ModelSerializer.Deserialize will never return null in this case?  Is it because it would throw an exception?